### PR TITLE
defaultDocument implementation

### PR DESCRIPTION
**Pull request by boemekeld**
Hey guys,

In some cases, "index.html" is not the default document.
I work on some projects that are being migrated to Edge and that use "index.shtm" for example.
For this, I suggest the inclusion of an option **defaultDocument**.

Thanks. 👍

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1 @@
-node_modules
-dist
```
Comment (SukkaW) [inline .gitignore:2]: Why remove `dist` from `.gitignore`?

**Review by SukkaW — CHANGES_REQUESTED** [review]

**Comment by SukkaW** [comment]
> I work on some projects that are being migrated to Edge and that use "index.shtm" for example.

`shtml` is not "static HTML". It is "server-side HTML" which means the served file is generated dynamically.

File: .gitignore (hunk 1 above)
Comment (boemekeld) [reply .gitignore]: Hi @SukkaW, my mistake. Was solved. Thanks.

**Comment by boemekeld** [comment]
> website

I am building a platform that can host different stacks, including websites with SSI (Server Side Includes).

There are a number of reasons that do not allow me to rename the files to .HTML, like as SEO, compatibility with other environments, etc.

In this necenary, SHTML files are previously generated in the build stage, so they are static files.

I appreciate your comment. Thanks.

**Comment by boemekeld** [comment]
Hi @SukkaW

Could you review this pull request?

Thanks.

**Comment by boemekeld** [comment]
Up

**Comment by boemekeld** [comment]
@signalnerve of course! I will do this soon.

**Comment by boemekeld** [comment]
@signalnerve done!

**Comment by Cherry** [comment]
Thanks, this looks like a good change and should close #87.

@boemekeld Could you add a test for this new functionality for `mapRequestToAsset` in `src/test/mapRequestToAsset.ts`?

**Comment by boemekeld** [comment]
@Cherry yes, let me check it.

**Comment by boemekeld** [comment]
@Cherry done!

**Comment by signalnerve** [comment]
> @Cherry done!

we're going to work on supporting running our test workflow on pull requests, but in the meantime, can you post your test output here just so I have a paper trail before merge? ty!

**Comment by boemekeld** [comment]
@signalnerve sure.

> @cloudflare/kv-asset-handler@0.1.1 pretest C:\git\bohr-cloud\kv-asset-handler
> npm run build

> @cloudflare/kv-asset-handler@0.1.1 build C:\git\bohr-cloud\kv-asset-handler
> tsc -d

> @cloudflare/kv-asset-handler@0.1.1 test C:\git\bohr-cloud\kv-asset-handler
> ava dist/test/*.js --verbose

  - getAssetFromKV » getAssetFromKV when body not empty, should invoke .cancel()
  √ mapRequestToAsset » mapRequestToAsset() correctly changes /about -> /about/index.html
  √ mapRequestToAsset » mapRequestToAsset() correctly changes /about/ -> /about/index.html
  √ mapRequestToAsset » mapRequestToAsset() correctly changes /about.me/ -> /about.me/index.html
  √ mapRequestToAsset » mapRequestToAsset() correctly changes /about -> /about/default.html
  √ serveSinglePageApp » serveSinglePageApp returns root asset path when request path ends in .html
  √ serveSinglePageApp » serveSinglePageApp returns root asset path when request path does not have extension
  √ serveSinglePageApp » serveSinglePageApp returns requested asset when request path has non-html extension
Environment doesnt support readable streams
  √ getAssetFromKV » getAssetFromKV does not cache on Cloudflare when bypass cache set
Environment doesnt support readable streams
  √ getAssetFromKV » getAssetFromKV if not in asset manifest still returns nohash.txt
  √ getAssetFromKV » getAssetFromKV when setting browser caching
  √ getAssetFromKV » getAssetFromKV when namespace not bound fails
  √ getAssetFromKV » getAssetFromKV return correct val from KV and default caching
  √ getAssetFromKV » getAssetFromKV evaluated the file matching the extensionless path first /client/ -> client
  √ getAssetFromKV » getAssetFromKV evaluated the file matching the extensionless path first /client -> client
  √ getAssetFromKV » getAssetFromKV if no asset manifest /client -> client fails
  √ getAssetFromKV » getAssetFromKV if sub/ -> sub/index.html served
  √ getAssetFromKV » getAssetFromKV gets index.html by default for / requests
  √ getAssetFromKV » getAssetFromKV non ASCII path support
  √ getAssetFromKV » getAssetFromKV supports browser percent encoded URLs
  √ getAssetFromKV » getAssetFromKV supports user percent encoded URLs
  √ getAssetFromKV » getAssetFromKV custom key modifier
  √ getAssetFromKV » getAssetFromKV with no trailing slash on root
  √ getAssetFromKV » getAssetFromKV with no trailing slash on a subdirectory
  √ getAssetFromKV » getAssetFromKV no result throws an error
  √ getAssetFromKV » getAssetFromKV passing in a custom NAMESPACE serves correct asset
  √ getAssetFromKV » getAssetFromKV when setting custom cache setting
  √ getAssetFromKV » getAssetFromKV when custom namespace without the asset should fail
  √ getAssetFromKV » getAssetFromKV only decode URL when necessary
  √ getAssetFromKV » getAssetFromKV Support for user decode url path
  √ getAssetFromKV » getAssetFromKV when etag and if-none-match are present but if-none-match !== etag, should bypass cache
  √ getAssetFromKV » getAssetFromKV caches on two sequential requests
  √ getAssetFromKV » getAssetFromKV if-none-match not sent but resource in cache, should return hit
  √ getAssetFromKV » getAssetFromKV if range request submitted and resource in cache, request fulfilled
  √ getAssetFromKV » getAssetFromKV TTls set to null should not cache on browser or edge (111ms)
Environment doesnt support readable streams
Environment doesnt support readable streams
  √ getAssetFromKV » getAssetFromKV does not store max-age on two sequential requests (116ms)
  √ getAssetFromKV » getAssetFromKV when if-none-match === etag and etag === pathKey in manifest, should revalidate (108ms)
  ─

  36 tests passed
  1 test todo

**Review by Cherry — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**